### DISCUSSION TITLE
Selecting null area faces?

### DISCUSSION THREAD

**Discussion by Cervidellus** [cnr-isti-vclab/PyMeshLab#103] [Q&A] [answered]
I am trying to clean up some large and difficult meshes. The way we generate them produces a decent number of null-area faces. Is there a simple way of selecting null area faces without removing them? Most of them seem to get fixed by remove_t_vertices_by_edge_flip, but after doing this I want to check and make sure they are all fixed before moving on. In the GUI version, I am doing so by coloring the mesh by face area, and then selecting by face quality, but this feels a little awkward.

**Comment by alemuntoni** [comment, marked as answer]
Sorry for the very late reply!

You could use the [`per_face_quality_according_to_triangle_shape_and_aspect_ratio`](https://pymeshlab.readthedocs.io/en/latest/filter_list.html#per_face_quality_according_to_triangle_shape_and_aspect_ratio) filter, using `Area` as metric, to set the area into the quality field.
Then, you could use the `conditional_face_selection` filter with `condselect='fq==0'`, that will select all faces having quality 0.
Then, you can check the number of selected faces with the [`selected_face_number`](https://pymeshlab.readthedocs.io/en/latest/classes/mesh.html#pmeshlab.Mesh.selected_face_number) method of the Mesh class.
